bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/sitequality/SiteQualityPanel.java

```diff
@@ -166,43 +164,59 @@
 				weighty = 50;
 			}});
 
+            final Frame sqpFrame = (Frame) SwingUtilities.getWindowAncestor(this);
+
             add(new Box(BoxLayout.LINE_AXIS) {{
 				add(new JButton(Resources.getIcon("eclipse/add.gif")) {{
                     setToolTipText("Add new timing window");
                     setFocusable(false);
-                    addActionListener(e -> {
-						TimingWindow tw = new TimingWindow();
-						TimingWindowDialog twd = new TimingWindowDialog((Frame) SwingUtilities.getWindowAncestor(SiteQualityPanel.this));
-						tw = twd.showEdit(tw);
-						if (tw != null) {
+                    addActionListener(e ->
+						new TimingWindowDialog(sqpFrame).openNew().foreach(tw -> {
 							owner.getDataObject().addTimingWindow(tw);
 							int index = table.getModel().getRowCount() - 1;
-							table.changeSelection(index, 0, false, false); // magic
-						}
-					});
+							table.changeSelection(index, 0, false, false);
+						}));
                     ButtonFlattener.flatten(this);
 				}});
 
+                add(new JButton(Resources.getIcon("eclipse/download.gif")) {{
+                    setToolTipText("Import timing windows from text file");
+                    setFocusable(false);
+                    addActionListener(e -> {
+                        final TimingWindowImporter importer = new TimingWindowImporter(SiteQualityPanel.this);
+						final TimingWindowParser.TimingWindowParseResults results = importer.openImport();
+
+                        final int newIndex = table.getModel().getRowCount();
+                        owner.getDataObject().addTimingWindows(results.successesAsJava());
```

Comment:
Thanks! I figured it was important to indicate failures, although haha delving back into the world of creating `JTable`s instead of just modifying existing implemented ones may result in a few nights of nightmares!